packages/backend/src/types.ts

```diff
@@ -14,14 +14,16 @@
  * limitations under the License.
  */
 
-import Knex from 'knex';
 import { Logger } from 'winston';
 import { Config } from '@backstage/config';
-import { PluginEndpointDiscovery } from '@backstage/backend-common';
+import {
+  PluginDatabaseClientFactory,
+  PluginEndpointDiscovery,
+} from '@backstage/backend-common';
 
 export type PluginEnvironment = {
   logger: Logger;
-  database: Knex;
+  databaseClientFactory: PluginDatabaseClientFactory;
```

Comment:
As noted above this could be renamed back to just `database` if it were a `PluginDatabaseManager` type that has a `getDatabaseClient` method.

Reply:
Addressed in https://github.com/spotify/backstage/commit/35c0294018b8433787a795b7d9fcb6e9daef2b68.